### PR TITLE
SWC-7064 - Entity Upload: add Link to URL tab

### PR DESCRIPTION
**Pull request by nickgros**
image

### CODE REVIEW

File: packages/synapse-react-client/src/synapse-queries/entity/useEntity.ts [hunk 1]
```diff
@@ -46,21 +47,29 @@ import { getNextPageParamForPaginatedResults } from '../InfiniteQueryUtils'
...
+    queryOptions<T, SynapseClientError>({
+      queryKey: keyFactory.getEntityVersionQueryKey(entityId, versionNumber),
+      queryFn: () =>
+        SynapseClient.getEntity<T>(
+          accessToken,
+          entityId,
+          versionNumber?.toString(),
+        ),
+    })
+}
```
Comment (nickgros) [inline packages/synapse-react-client/src/synapse-queries/entity/useEntity.ts:62]: Extracted to allow inline fetch with `queryClient.fetchQuery(options)`

File: packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/useLinkFileEntityToURL.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      const firstQuestionMark = path.indexOf('?', lastSlash)
+      if (firstQuestionMark > -1) {
+        fileName = path.substring(lastSlash + 1, firstQuestionMark)
+      } else {
+        fileName = path.substring(lastSlash + 1)
+      }
+    }
+  }
+  return fileName
+}
```
Comment (nickgros) [inline packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/useLinkFileEntityToURL.ts:29]: copied from SWC

File: packages/synapse-react-client/src/components/EntityUpload/LinkToURL.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  const { data: entity } = useGetEntity(entityId)
+  const isFileEntity =
+    entity?.concreteType === 'org.sagebionetworks.repo.model.FileEntity'
+
+  const { mutate, error } = useLinkFileEntityToURL({
+    onSuccess,
+  })
+
+  useEffect(() => {
+    const isValid = url.trim().length > 0
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/components/EntityUpload/LinkToURL.tsx:40]: Any additional validation needed here?  I'm not sure how much sanitation we do on the rendering side for this URL value throughout the web apps.

Comment (jay-hodgson) [reply packages/synapse-react-client/src/components/EntityUpload/LinkToURL.tsx:40]: We can't assume this is set via the web interface, so I think this is fine.

Comment (nickgros) [reply packages/synapse-react-client/src/components/EntityUpload/LinkToURL.tsx:40]: This is the same behavior as SWC -- only check that the URL is a nonempty string and let the backend validate syntax

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**